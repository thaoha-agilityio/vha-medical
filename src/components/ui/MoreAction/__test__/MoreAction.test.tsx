import { render } from '@testing-library/react';
import { MenuAction, MoreActionProps } from '..';

describe('MoreAction test cases', () => {
  const setup = (props: MoreActionProps) => render(<MenuAction {...props} />);

  const mockAction = jest.fn();
  const mockProps: MoreActionProps = {
    options: [
      {
        key: '1',
        label: 'Mock',
        onAction: mockAction,
        isDisabled: false,
      },
    ],
  };
  it('should render correctly', () => {
    const { asFragment } = setup(mockProps);

    expect(asFragment()).toMatchSnapshot();
  });
});
