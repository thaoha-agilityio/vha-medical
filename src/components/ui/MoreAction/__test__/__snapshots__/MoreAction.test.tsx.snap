// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreAction test cases should render correctly 1`] = `
<DocumentFragment>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    aria-label="Dropdown button"
    class="group relative items-center box-border appearance-none select-none whitespace-nowrap font-normal overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover z-10 aria-expanded:scale-[0.97] aria-expanded:opacity-70 subpixel-antialiased text-primary-100 border-1 data-[focus=true]:border-1 px-[2px] md:px-1 min-w-4 h-4 md:h-[26px] md:min-w-[26px] bg-background-100 rounded md:rounded-md data-[focus=true]:border-transparent data-[focus=true]:border-opacity-20 border-transparent border-opacity-20 flex justify-start capitalize"
    data-slot="trigger"
    id="react-aria-:r0:"
    type="button"
  >
    <div
      class="text-primary-100 max-h-6 max-w-6"
    >
      <svg
        class="w-[11px] h-[11px] md:w-4 md:h-4"
        fill="none"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 6C6.90076 6 6 6.90076 6 8C6 9.09924 6.90076 10 8 10C9.09924 10 10 9.09924 10 8C10 6.90076 9.09924 6 8 6Z"
          fill="currentColor"
        />
        <path
          d="M8 0C6.90076 0 6 0.900764 6 2C6 3.09924 6.90076 4 8 4C9.09924 4 10 3.09924 10 2C10 0.900764 9.09924 0 8 0Z"
          fill="currentColor"
        />
        <path
          d="M8 12C6.90076 12 6 12.9008 6 14C6 15.0992 6.90076 16 8 16C9.09924 16 10 15.0992 10 14C10 12.9008 9.09924 12 8 12Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;
